src/Magnum/LibOVRIntegration/Test/IntegrationTest.cpp

```diff
@@ -0,0 +1,209 @@
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <Corrade/TestSuite/Tester.h>
+
+#include "Magnum/Magnum.h"
+#include "Magnum/LibOvrIntegration/Conversion.h"
+#include "Magnum/Math/Matrix3.h"
+#include "Magnum/Math/DualQuaternion.h"
+
+namespace Magnum { namespace LibOvrIntegration { namespace Test {
+
+// define missing == operators for LibOVR types
+bool operator== (const ovrSizei& a, const ovrSizei& b);
+bool operator== (const ovrVector2i& a, const ovrVector2i& b);
+bool operator== (const ovrVector2f& a, const ovrVector2f& b);
+bool operator== (const ovrVector3f& a, const ovrVector3f& b);
+bool operator== (const ovrRecti& a, const ovrRecti& b);
+bool operator== (const ovrMatrix4f& a, const ovrMatrix4f& b);
+bool operator== (const ovrQuatf& a, const ovrQuatf& b);
+bool operator== (const ovrPosef& a, const ovrPosef& b);
+
+bool operator== (const ovrSizei& a, const ovrSizei& b) {
+    if (a.w != b.w) return false;
+    if (a.h != b.h) return false;
+    return true;
+}
+
+bool operator== (const ovrVector2i& a, const ovrVector2i& b) {
+    if (a.x != b.x) return false;
+    if (a.y != b.y) return false;
+    return true;
+}
+
+bool operator== (const ovrVector2f& a, const ovrVector2f& b) {
+    if (a.x != b.x) return false;
+    if (a.y != b.y) return false;
+    return true;
+}
+
+
+bool operator== (const ovrVector3f& a, const ovrVector3f& b) {
+    if (a.x != b.x) return false;
+    if (a.y != b.y) return false;
+    if (a.z != b.z) return false;
+    return true;
+}
+
+bool operator== (const ovrRecti& a, const ovrRecti& b) {
+    if (!(a.Pos == b.Pos)) return false;
+    if (!(a.Size == b.Size)) return false;
+    return true;
+}
+
+bool operator== (const ovrMatrix4f& a, const ovrMatrix4f& b) {
+    float *pa = (float*) a.M,
+          *pb = (float*) b.M;
+
+    for (int i = 0; i < 16; ++i, ++pa, ++pb) {
+        if(*pa != *pb)
+            return false;
+    }
+
+    return true;
+}
+
+bool operator== (const ovrQuatf& a, const ovrQuatf& b) {
+    if (a.x != b.x) return false;
+    if (a.y != b.y) return false;
+    if (a.z != b.z) return false;
+    if (a.w != b.w) return false;
+    return true;
```

Comment:
Well, thinking about these, I think it might be better to drop the operators and do the comparisons directly in the test cases, as it would do proper fuzzy comparison for floats, will verbosely print what's wrong when the test fails and won't break if next release of Oculus SDK adds the comparison operators itself. So in the test case below:

```cpp
ovrQuatf c{a};
CORRADE_COMPARE(c.x, b.x);
CORRADE_COMPARE(c.y, b.y);
CORRADE_COMPARE(c.z, b.z);
CORRADE_COMPARE(c.w, b.w);
```

You can call the `CORRADE_COMPARE` macro in a loop, so that would work fine also in the matrix case.